util/llbutil/copyop.go

```diff
@@ -30,17 +30,12 @@ func CopyOp(ctx context.Context, srcState pllb.State, srcs []string, destState p
 		baseCopyOpts = append(baseCopyOpts, llb.WithCreatedTime(*defaultTs()))
 	}
 	for _, src := range srcs {
-		// If the copy came in as optional (ifExists), then we need to trigger
-		// the underlying wildcard matching and allow empty wildcards. The
-		// matching uses the filepath match syntax. So by simply creating a
-		// wildcard where the first letter needs to match the current first
-		// letter gets us the single match; and no error if it is missing. Paths
-		// with wildcards will be automatically ignored when no match is present.
-		if ifExists && len(src) != 0 && !strings.Contains(src, "*") {
-			// Normalize path by dropping './'
+		// HACK: For COPY --if-exists, we can use a glob expression (e.g., '[f]oo') to
```

Comment:
I'm confused by the comment -- where are we using a glob expression?

Reply:
```
src = fmt.Sprintf("[%s]%s", string(src[0]), string(src[1:]))
```
This code uses a bit of a hack to have BK treat the file(s) as optional. `foo` becomes `[f]oo`, which has BK interpret the source as an expression to expand.

Reply:
oh wow.... yeah I didn't understand what that particular line was doing. This is really hacky.

A comment directly above this line would make it easier, maybe something like:

```
// HACK: a regular expression is constructed, e.g. "foo" becomes "[f]oo"
src = fmt.Sprintf("[%s]%s", string(src[0]), string(src[1:]))
```

Reply:
Sure, I moved the comment.

Reply:
thanks, it feels easier to follow now.